src/core/histogram.cc

```diff
@@ -96,6 +96,34 @@ histogram::print(FILE *fp) const {
                                  CLASSIC);  // Format CLASSIC/CSV supported.
 }
 
+seastar::metrics::histogram
+histogram::get_seastar_metrics_histogram() const {
+  seastar::metrics::histogram sshist;
+  struct hdr_iter iter;
+  struct hdr_iter_percentiles *percentiles;
+
+  hdr_iter_percentile_init(&iter, hist_->hist, 5);
+  percentiles = &iter.specifics.percentiles;
+
+  uint64_t running_total = 0;
+  while (hdr_iter_next(&iter))
+  {
+    double  value               = iter.highest_equivalent_value / 1.0;
+    double  percentile          = percentiles->percentile / 100.0;
+    int64_t total_count         = iter.cumulative_count;
+    struct seastar::metrics::histogram_bucket bucket;
+    bucket.count = total_count - running_total;
+    running_total += bucket.count;
+    bucket.upper_bound = value;
+    sshist.buckets.push_back(bucket);
```

Comment:
compute the reservaction on the buckets and call `vector::reserve()` to make sure you don't realloc multiple times in this method.